src/main/java/com/matt/forgehax/mods/services/FriendService.java

```diff
@@ -0,0 +1,205 @@
+          message.replace(f.getName(), TextFormatting.RESET.toString() + TextFormatting.LIGHT_PURPLE + 
+                                                f.getName() + TextFormatting.RESET + format_pre));
+        event.setMessage(out);
+        break;
+      }
+    }
+  }
+
+  @SubscribeEvent
+  public void onTabUpdate(RenderTabNameEvent event) {
+    if (!color_tab.get()) return;
+
+    FriendEntry f = friendList.stream()
+          .filter(item -> item.getName().equals(event.getName()))
+          .findFirst()
+          .orElse(null);
+    if (f != null)
+      event.setColor(Colors.GOLD.toBuffer());
+  }
+
+  @Override
+  protected void onLoad() {
+    super.onLoad();
+    friendList.deserializeAll();
+    
+    getCommandStub()
+        .builders()
+        .newCommandBuilder()
+        .name("add")
+        .description("Add new friend (by name)")
+        .processor(
+            data -> {
+              data.requiredArguments(1);
+              final String name = data.getArgumentAsString(0);
+              try {
+                this.friendList.add(new FriendEntry(name));
+                Helper.printMessage("Added \"%s\" as friend", name);
+              } catch (Exception ex) {
+                Helper.printError("Could not add a friend for some reason (:");
+              }
+            })
+        .build();
+
+    getCommandStub()
+        .builders()
+        .newCommandBuilder()
+        .name("remove")
+        .description("Remove a friend (by Name)")
+        .processor(
+            data -> {
+              data.requiredArguments(1);
+              final String name = data.getArgumentAsString(0);
+              final boolean changed = friendList.removeIf(entry -> entry.name.equals(name));
+              if (changed) {
+                Helper.printMessage("Removed \"%s\" from friends", name);
+              } else {
+                Helper.printInform("No friend named \"%s\" was found", name);
+              }
+            })
+        .build();
+
+    getCommandStub()
+        .builders()
+        .newCommandBuilder()
+        .name("list")
+        .description("List all friends")
+        .processor(data -> {
+          Helper.printMessage("%d friends", friendList.size());
+          for (FriendEntry entry : friendList) {
+            data.write(entry.name);
+          }
+        })
+        .build();
+  }
+
+  @Override
+  public void onUnload() {
+    friendList.serialize();
+  }
+
+
+  public static class FriendEntry implements ISerializableJson {
+    private String name;
+    // TODO identify by UUID
+
+    FriendEntry(String name) {
+      this.name = name;
+    }
+
+    public String getName() {
+      return this.name;
+    }
+
+    @Override
+    public void serialize(JsonWriter writer) throws IOException {
+      writer.value(this.name);
+    }
+
+    @Override
+    public void deserialize(JsonReader reader)  {
+      this.name  = new JsonParser().parse(reader).getAsString();
```

Comment:
wtf does this actually work?

Reply:
It seems like it does, why shouldn't it? I tried to have as little to do as possible with `Json` in java, but this is the only way I found